apps/site/src/components/save-on-blur-text-field.tsx

```diff
@@ -0,0 +1,107 @@
+import { faCheck } from "@fortawesome/free-solid-svg-icons";
+import {
+  Box,
+  CircularProgress,
+  outlinedInputClasses,
+  TextFieldProps,
+  useTheme,
+} from "@mui/material";
+import { AnimatePresence, m } from "framer-motion";
+import { FocusEvent, useState } from "react";
+
+import { FontAwesomeIcon } from "./icons";
+import { TextField } from "./text-field";
+
+type SaveOnBlurTextFieldProps = Omit<TextFieldProps, "onBlur"> & {
+  onSave: (value: string) => Promise<void>;
+};
+
+const outlineSelector = `.${outlinedInputClasses.notchedOutline}`;
+
+export const SaveOnBlurTextField = ({
+  onSave,
+  onChange,
+  value,
+  ...rest
+}: SaveOnBlurTextFieldProps) => {
+  const theme = useTheme();
+
+  const [isLoading, setIsLoading] = useState(false);
+  const [savedAtLeastOnce, setSavedAtLeastOnce] = useState(false);
+  const [lastSavedVal, setLastSavedVal] = useState(value);
+
+  const handleBlur = async (event: FocusEvent<HTMLInputElement>) => {
+    const newValue = event.target.value;
+    const notChanged = newValue === lastSavedVal;
+
+    if (notChanged) {
+      return;
+    }
+
+    try {
+      setIsLoading(true);
+      setSavedAtLeastOnce(true);
```

Comment:
This will make a checkmark appear even if the save fails – I think we want to set it after `onSave`?